apps/react/src/challenges/15puzzle/components/Board/index.tsx

```diff
@@ -0,0 +1,190 @@
+import { AnimationDirection } from '@/challenges/15puzzle/AnimationDirection.ts';
+import Tile from '@/challenges/15puzzle/components/Tile';
+import { useCallback, useEffect, useState } from 'react';
+import styles from './board.module.css';
+
+interface Poz {
+  row: number;
+  col: number;
+}
+
+interface PrevMoves {
+  moves: Poz[];
+  capacity: number;
+}
+
+interface Animation {
+  element: Poz | null;
+  type: AnimationDirection | '';
+}
+
+export default function Board() {
+  const [grid, setGrid] = useState<number[][]>([]);
+  const [totalMoves, setTotalMoves] = useState(0);
+  const [isGridSorted, setIsGridSorted] = useState(false);
+  const [prevMoves, setPrevMoves] = useState<PrevMoves>({ moves: [], capacity: 5 });
+  const [animation, setAnimation] = useState<Animation>({ element: null, type: '' });
+
+  const addPrevMove = (move: Poz) => {
+    const { moves, capacity } = prevMoves;
+    if (moves.length === capacity) moves.shift();
+
+    moves.push(move);
+  };
+
+  const isSolvable = (grid: number[]) => {
+    let invCount = 0;
+
+    for (let i = 0; i < 15; i++) {
+      for (let j = i + 1; j < 16; j++) {
```

Comment:
15 and 16 need not be hard coded. Extending will become challenge

Reply:
Yea, that's a good point, I planned to work on it, but your refactor looking nice. Thanks